src/main/java/com/google/sps/data/DeliverySlotManager.java

```diff
@@ -0,0 +1,201 @@
+      } catch (ParseException e) {
+        throw new BadRequestException("Please enter a valid date!");
+      }
+      // get the currentDate to check that date is not set for a past date
+      Date currentDay = new Date(System.currentTimeMillis());
+      if (date.compareTo(currentDay) < 0) {
+        throw new BadRequestException("Please enter a valid date!");
+      }
+      if (date.getTime() + startTimeSeconds * 1000 < currentDay.getTime()) {
+        throw new BadRequestException("Please enter a time in the future!");
+      }
+      return date;
+    }
+
+    /**
+     * Returns the integer value of numberString
+     */
+    private Integer parseInteger(String numberString) throws BadRequestException {
+      Integer number = null;
+      try {
+        number = Integer.parseInt(numberString);
+      } catch (Exception e) {
+        throw new BadRequestException(e.getMessage());
+      }
+      return number;
+    }
+
+    /**
+     * Returns the Double value of numberString
+     */
+    private Double parseDouble(String numberString) throws BadRequestException {
+      Double number = null;
+      try {
+        number = Double.parseDouble(numberString);
+      } catch (Exception e) {
+        throw new BadRequestException(e.getMessage());
+      }
+      return number;
+    }
+
+    /**
+     * Returns LocalTime object from timeString
+     */
+    private LocalTime parseTime(String timeString) throws BadRequestException {
+      LocalTime time = null;
+      DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm");
+      try {
+        time = LocalTime.parse(timeString, formatter);
+      } catch (DateTimeParseException e) {
+        throw new BadRequestException(e.getMessage());
+      }
+      return time;
+    }
+
+    private int getNumberOfSeconds(LocalTime time) {
+      return time.getHour() * 3600 + time.getMinute() * 60;
+    }
+  }
+
+  private CreateSlotManager slotRequest;
+  private String userId;
+  
+  /** 
+   * Creates a delivery slot request given the delivery day, the timezone offset in minutes, start
+   * and end times of the delivery, the maximum number of stops and user's id.
+   */
+  public void createSlot(String deliveryDay, String timezoneOffsetMinutes, String startTime, String endTime, String maxStops, String userId) throws BadRequestException {
+    this.slotRequest = new CreateSlotManager(deliveryDay, timezoneOffsetMinutes, startTime, endTime, maxStops);
+    this.userId = userId;
+  }
+
+  public void setSlotStartingPoint(String address) throws ApiException, InterruptedException, IOException, DataNotFoundException {
+    this.slotRequest.setCoordinatesFromAddress(address);
+  }
+  public void setSlotStartingPoint(String latitude, String longitude) throws BadRequestException {
+    this.slotRequest.setCoordinates(latitude, longitude);
+  }
+  public DeliverySlot getDeliverySlot() {
+    return slotRequest.deliverySlot;
+  }
+  public double getStartLatitude() {
```

Comment:
Really don't add getters/setters unless they are needed by anything. They are just noise at this point. It's always a good idea to have a smaller public surface and to keep most things private when possible.